### DISCUSSION TITLE
bagaimana cara import kata-dasar.txt di jupyter notebook

### DISCUSSION THREAD

**Discussion by Gustana** [teamalgoritma/community#630] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/23212456/231705145-9ff30b84-5418-4d60-9c10-39fbe51a75c6.png)
![image](https://user-images.githubusercontent.com/23212456/231705361-0c466e8a-e5e6-4620-ac6c-f0b30d22875d.png)

**Comment by amaliasf** [comment]
Dear Mas @Gustana 

Untuk menjalankan stemmer dari Python Sastrawi dan menggunakan kata-dasar.txt, berikut langkah yang bisa dilakukan:

**1. Import StemmerFactory class**
`from Sastrawi.Stemmer.StemmerFactory import StemmerFactory`

**2. Create stemmer**
```
factory = StemmerFactory()
stemmer = factory.create_stemmer()
```

**3. Stemming Process**
`print(stemmer.stem('kalimat yang ingin diterapkan fungsi stem'))`

Maka proses stemming sudah berhasil dilakukan. Untuk dokumentasi lengkap dari Python Sastrawi bisa merujuk pada [Python Sastrawi](https://github.com/har07/PySastrawi).

Semoga informasi ini dapat membantu. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Comment by Gustana** [comment]
saya tidak ingin melakukan stem, saya ingin mendapatkan list yang berisi kata-dasar.txt dari library sastrawi tanpa mendefinisikan manual kata dasar tersebut di code, apakah itu memungkinkan?

**Comment by hwulanayu** [comment, marked as answer]
Dear @Gustana,

Cara termudah adalah dengan download txt filenya langsung di https://github.com/har07/PySastrawi/blob/master/src/Sastrawi/Stemmer/data/kata-dasar.txt dan read di Python seperti biasa. Cara kedua adalah dengan menggunakan library [pkg_resources](https://setuptools.pypa.io/en/latest/pkg_resources.html):
```
import pkg_resources
kata_dasar_file = pkg_resources.resource_filename("Sastrawi", 'Stemmer/data/kata-dasar.txt')
with open(kata_dasar_file) as file:
    data = file.read().split('\n')


print(len(data))
print(type(data))
data
```
